pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -495,19 +507,21 @@ TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testTrajCommandExecution)
       nh.advertise<trajectory_msgs::JointTrajectory>(CONTROLLER_NAMESPACE + TRAJECTORY_COMMAND_TOPIC, 1);
 
   GoalType goal{ generateSimpleGoal() };
-  trajectory_command_publisher.publish(goal.trajectory);
-
-  EXPECT_TRUE(updateUntilRobotMotion<RobotDriver>(&robot_driver_));
 
-  bool is_executing_result;
-  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_TRUE(is_executing_result) << "Failed to detect trajectory command execution";
+  pilz_testutils::ROSLogExtender ros_log_extender;
+  EXPECT_CALL(*ros_log_extender,
+              append(IsWarn("For safety reasons the trajectory command interface is deactivated "
+                            "(for more information see https://github.com/ros-controls/ros_controllers/issues/493). "
```

Comment:
Why test for specific message? Imho it would be enough to test, _if_ there is a warning and the robot doesn't move.

Reply:
If things go wrong (hopefully) the will be warnings. But we try to check for a specific kind of "wrong" to avoid false positives.

Reply:
@jschleicher In my opinion, your statements contradict each other. On the one hand you say that the link is not good, on the other hand, the message does not count, only that it is a warning. What is the  overall logging concept/idea? What should and what should not be stated?

Reply:
How do ensure that the user get's the best possible feedback, in case of errors/problems?

Reply:
The error concept should be solution driven. Git command line has the concept "maybe you mean xy?" suggesting possible options. Instead of explaining hard-to-follow reasons, I'd suggest to provide solutions - in our case refer to the action interface. Linktip: [The message should help the user solve the problem and move on.](https://medium.com/s/user-friendly/the-art-of-the-error-message-9f878d0bff80)
But since this semantics cannot be tested here anyways, the exact formulation shouldn't be duplicated within the test - this only leads to increased maintenance effort, even in case of typos you have to touch two distinct files.

Reply:
I think pointing to the solution is a good idea.

But in think my argument that checking for the right message is needed to avoid false positives in the test  is not countered yet. The duplication could/should be removed by defining the precise message at a single location.

Reply:
Another solution would be for the user to close https://github.com/ros-controls/ros_controllers/issues/493
PR welcome :wink: 
"If you want to use this interface please fix ... first! PR welcome ;-)"

Reply:
> The error concept should be solution driven. Git command line has the concept "maybe you mean xy?" suggesting possible options. Instead of explaining hard-to-follow reasons, I'd suggest to provide solutions - in our case refer to the action interface. Linktip: [The message should help the user solve the problem and move on.](https://medium.com/s/user-friendly/the-art-of-the-error-message-9f878d0bff80)
> But since this semantics cannot be tested here anyways, the exact formulation shouldn't be duplicated within the test - this only leads to increased maintenance effort, even in case of typos you have to touch two distinct files.

https://github.com/PilzDE/pilz_robots/pull/405/commits/c1fa6d7c7dd68bd8eabc3f06d22157eddcd5dd7d
Duplication removed -> maintenance effort reduced.

Reply:
Also outdated since messages have been adapted.